src/proxy/http/router.rs

```diff
@@ -25,17 +25,14 @@ pub struct Config {
 /// A `Rec`-typed `Recognize` instance is used to produce a target for each
 /// `Req`-typed request. If the router doesn't already have a `Service` for this
 /// target, it uses a `Stk`-typed `Service` stack.
-#[derive(Debug)]
-pub struct Layer<Req, Rec, Stk> {
+#[derive(Clone, Debug)]
+pub struct Layer<Req, Rec: Recognize<Req>> {
     recognize: Rec,
-    _p: PhantomData<fn() -> (Req, Stk)>,
+    _p: PhantomData<fn() -> Req>,
```

Comment:
I _think_ this phantomdata is unnecessary because of the `Rec: Recognize<Req>` bound (but I could be wrong).

Reply:
phanomdata is necessary.